src/crypto-api/index.ts

```diff
@@ -1243,6 +1250,72 @@ export interface OwnDeviceKeys {
     curve25519: string;
 }
 
+export enum DehydratedDevicesEvents {
+    /** Emitted when a new dehydrated device is created locally */
+    DeviceCreated = "DeviceCreated",
+    /** Emitted when a new dehydrated device is successfully uploaded to the server */
+    DeviceUploaded = "DeviceUploaded",
+    /** Emitted when rehydration has started */
+    RehydrationStarted = "RehydrationStarted",
+    /** Emitted when rehydration has finished */
+    RehydrationEnded = "RehydrationEnded",
+    /** Emitted during rehydration, signalling the current `roomKeyCount` and `toDeviceCount` */
+    RehydrationProgress = "RehydrationProgress",
```

Comment:
this needs better documentation on what the payload of the event is.

```suggestion
    /**
      * Emitted during rehydration, to inform the application of rehydration progress.
      *
      * The payload is a pair `[roomKeyCount: number, toDeviceCount: number]`, where
      * `roomKeyCount` is (what, exactly??) and ``toDeviceCount` is (something else??
      */
    RehydrationProgress = "RehydrationProgress",
```